packages/@vue/cli-plugin-router/generator/template/src/router/index.js

```diff
@@ -1,9 +1,35 @@
 import Vue from 'vue'
+<%_ if (hasTypeScript) { _%>
+import VueRouter, { RouteConfig } from 'vue-router'
+<%_ } else { _%>
 import VueRouter from 'vue-router'
+<%_ } _%>
 import Home from '../views/Home.vue'
 
-Vue.use(VueRouter)
+<%_ if (hasTypeScript) { _%>
+const routes = <Array<RouteConfig>>[
```

Comment:
As far as I can see, only this line needs to be modified for typescript projects. So no need to  wrap the whole block in the `if` condition.

Reply:
Better use type annotation than type casting:
```suggestion
const routes: Array<RouteConfig> = [
```

Reply:
Adding this type annotation just bug the CLI and an error will popup